src/broker.js

```diff
@@ -338,4 +339,106 @@ function handleTokenAndAuthResponses (messageObject, clientId) {
                 break;
         }
     }
+}
+
+/**
+ *
+ * @param clientId
+ * @param username
+ * @param password
+ * @return {Promise}
+ */
+function brokerAuthenticationHandler (clientId, username, password) {
+    return new Promise((resolve, reject) => {
+        if (!wsManager.hasKey(clientId)) {
+            if (username && password) {
+                wsManager.createTokens(clientId, username, password.toString())
+                    .then(({ accessToken, refreshToken }) => wsManager.authenticate(clientId, accessToken))
+                    .then(() => process.nextTick(() => resolve()))
+                    .catch((err) => process.nextTick(() => {
+                        wsManager.close(clientId);
+                        reject(err);
+                    }));
+            } else {
+                resolve();
+            }
+        } else {
+            reject(new Error(`User with id ${clientId} already connected`));
+        }
+    });
+}
+
+/**
+ *
+ * @param clientId
+ * @param topic
+ * @return {Promise}
+ */
+function brokerAuthorizeSubscriptionHandler (clientId, topic) {
```

Comment:
Is there any way to refactor this a bit?

Reply:
It was decided to leave this part as is and return to its' possible refactoring later